### DISCUSSION TITLE
Mesh partitioning optimization problem

### DISCUSSION THREAD

**Discussion by nuomi68** [idaholab/moose#27235] [Q&A General]
Hello everyone！
I have observed that mesh partitioning in MOOSE is significantly faster compared to using LibMesh directly. I'm curious to understand the optimizations or enhancements made within MOOSE that contribute to this speedup in mesh partitioning. Are these optimizations something that can be leveraged solely within LibMesh?

My current problem-solving cycle is quite complex and challenging to implement directly within the MOOSE framework; otherwise, I would be inclined to utilize MOOSE for my simulations. I'm specifically interested in whether there are ways to achieve similar partitioning speed improvements by tweaking settings or using certain features in LibMesh directly. Currently, I'm using the default Metis partitioner provided by LibMesh.

Any insights or suggestions on how to optimize mesh partitioning speed within LibMesh or details about the enhancements made in MOOSE would be greatly appreciated.

**Comment by GiudGiud** [comment]
Hello

Which partitioner are you interested about in MOOSE?
Some are very simple so they run very fast.

DId you read the partitioning doco?
https://mooseframework.inl.gov/syntax/Mesh/Partitioner/index.html
